### PR TITLE
Create Animal Sightings Tracker that utilizes SQL Joins and Forms for data management

### PR DESCRIPTION
**Pull request by cathydao95**
**Week 10 Project: Animal Sightings Tracker**

**Data you will work with:**
Species: This app will store data about different endangered species (e.g. polar bears, tigers). 
Individuals: Scientists track some individual animals of endangered species
Sightings: When scientists spot an species they’re tracking

**Things to Include:**
Error handling
Display some data from a min of 2 tables joined with SQL 
Show a list of all sightings, including the nickname of the individual sighted at each one (using a JOIN query).
Form to add a new sighting record
Form to add a new individual
Form to add a new species

https://github.com/cathydao95/animal-sighting-tracker/assets/79618165/2ac65a7f-2197-4e8f-8020-4213d426956c

### CODE REVIEW

**Comment by rkomorn** [comment]
What's going on with this super short PR title though?

File: client/src/App.jsx [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+    <BrowserRouter>
+      <div className="title">
+        <Link to="/">Animal Sighting Tracker</Link>
+      </div>
+
+      <Routes>
+        <Route path="/" element={<AllSpecies />}></Route>
+        <Route path="/:id" element={<IndividualSpecies />}></Route>
+        <Route
+          path="/individuals/:id/sighting"
```
Comment (rkomorn) [inline client/src/App.jsx:18]: I thought this might have the same problem you ran into in another project (where it was accidentally picking up the whole path as the `id`), but then I realized that was something in Express, not React.

File: client/src/components/Individuals/IndividualSpecies.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  ) : individualsOfSpecies.length > 0 ? (
+    <div>
+      {individualsOfSpecies && (
+        <h2>{`List of Individual ${individualsOfSpecies[0].common_name}s`}</h2>
+      )}
+      <IndividualsForm setIndividualsOfSpecies={setIndividualsOfSpecies} />
+      {individualsOfSpecies.length > 0 && (
+        <div className="container">
+          {individualsOfSpecies.map((individual) => {
+            const {
```
Comment (rkomorn) [inline client/src/components/Individuals/IndividualSpecies.jsx:45]: I think this could've been factored out into another component, just to look a little cleaner here.

File: client/src/components/Individuals/IndividualsForm.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      console.error("Error occured while creating event", error);
+    }
+    setIndividualInfo({
+      nickname: "",
+      scientist_email: "",
+      scientist_name: "",
+    });
+  };
+
+  console.log(individualInfo);
```
Comment (rkomorn) [inline client/src/components/Individuals/IndividualsForm.jsx:50]: Psst! You forgot this.

File: client/src/components/Sightings/IndividualSighting.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    } catch (error) {
+      console.error("Error occured while fetching data", error);
+    }
+  };
+
+  useEffect(() => {
+    getIndividualsSighting(id);
+  }, [id]);
+
+  console.log("here", individualsSightings, id);
```
Comment (rkomorn) [inline client/src/components/Sightings/IndividualSighting.jsx:32]: Near, far... Whereeeeever you are...

File: server/db.sql [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+5	Polar Bear	Ursus maritimus	22000	VU	2023-09-20 20:28:11.297703
+6	Panda	Ailuropoda melanoleuca	1800	EN	2023-09-20 20:28:11.297703
+\.
+
+
+--
+-- Name: individuals_id_seq; Type: SEQUENCE SET; Schema: public; Owner: tpl1122_1
+--
+
+SELECT pg_catalog.setval('public.individuals_id_seq', 19, true);
```
Comment (rkomorn) [inline server/db.sql:222]: Why is this necessary? What happens if you miss it?

**Review by rkomorn — COMMENTED** [review]
Posthumously approved.

File: server/db.sql (hunk 5 above)
Comment (cathydao95) [reply server/db.sql:222]: This is an automated file that is made for me :D

File: client/src/components/Individuals/IndividualsForm.jsx (hunk 3 above)
Comment (cathydao95) [reply client/src/components/Individuals/IndividualsForm.jsx:50]: Thank you for catching this. Will remove the unnessary console.logs!

File: client/src/components/Individuals/IndividualSpecies.jsx (hunk 2 above)
Comment (cathydao95) [reply client/src/components/Individuals/IndividualSpecies.jsx:45]: Oh good point! Done.

File: server/db.sql (hunk 5 above)
Comment (rkomorn) [reply server/db.sql:222]: Yes, I know, but it's still an interesting line to know about. :)
